### ISSUE TITLE
Allow deprecation of connection fields

### ISSUE THREAD

**Issue by MattGson** [graphql-nexus/schema#512]
There is currently no way to deprecate a field created with the connectionField plugin.

This is pretty important for maintaining versionless schemas.

This should work exactly like it does for any other field:

```ts
t.connectionField('some_field', {
    deprecation: 'No longer safe to use',
    type: 'SomeType',
    nodes: async (root, _, ctx) => await root.genMyFields(ctx))
});
```

**Comment by tgriesser** [comment]
Merged this in #578 - should be out with a release soon

**Comment by tgriesser** [comment]
Released in [0.17.0](https://github.com/graphql-nexus/schema/releases/tag/v0.17.0)
